client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the runtime context logging tools
+"""
+
+from concurrent import futures
+import unittest
+import pytest
+from typing import Any, Dict
+from verta import runtime
+
+
+def test_thread_safety() -> None:
```

Comment:
I think the true test we need here is whether `ctx.logs()` returns the logs after everything is done, and I'm not sure that it will with our current implementation because it can't read the logs from child threads 🤔

Reply:
I think perhaps we still want this to make sure the thread local variable for storing the logs is actually thread local and functions as expected.  However, we could push the test method called by each thread back one layer of scope, so to speak, so that each thread creates it's own context manager and verifies the content in `ctx.logs()` independently.  Is that kind of what you're thinking?

Reply:
Ok, check out the new `test_thread_safe_context` function below.  I believe it does what you're thinking.  I tried adding a bunch of random sleeps throughout the inner `log_in_context` function to ensure randomized timing of each thread and even though it makes the test really slow, it still passes.

Reply:
Ok, I was able to get the tests to work reliably.